dungeonGen.js

```diff
@@ -0,0 +1,284 @@
+$(function() {
+'use strict';
+
+	var sizeOpts, 
+		sizeChoice,
+		tSize,
+		points, 
+		posit, 
+		tile = $('canvas')[0].getContext('2d'), 
+		tileOut = $('canvas')[0].getContext('2d'),
+		width = $('canvas').width(), 
+		x = width/2,
+		height = $('canvas').height(), 
+		y = height/2,
+		coords = {"xPos":x, "yPos":y}; //default coordinates
+
+	//the initial black background
+	tile.fillStyle = "rgb(0,0,0)";
+	tile.fillRect(0, 0, width, height);
+
+	//disable the editing buttons upon app launch
+	$('#addButton').prop('disabled', true);
+	$('#undoButton').prop('disabled', true);
+
+	//Define a tile object
+	function Tile() {
+		this.inStack = [];
+		this.tileSize = tSize;
+
+		//find the index of the input coordinates in the tile array inStack
+		this.whereInStack = function(posit) {
+			for(var i = 0; i < this.inStack.length; i ++){
+				if(this.inStack[i][0] == posit[0] && this.inStack[i][1] == posit[1])
+					return i;
+			}
+			return -1;
+		}
+	}
+
+	//dungeon floor tile
+	function floorTile(color) {
+		this.color = color;
+		this.tileType = 'floor';
+	}
+
+	//setting the pointer to the dungeon floor tile's prototype to be the tile object
+	floorTile.prototype = new Tile();
+
+	//use this to keep track of the current dungeon tiles
+	var dungeonTiles = new floorTile('rgb(255,255,255)');
+
+	function drawTile(coords, size, color) {
+		tile.fillStyle = color;
+		tile.fillRect(coords[0], coords[1], size, size);
+		tileOut.strokeRect(coords[0], coords[1], size, size);
+	}
+
+	function addTiles(sizeOpts, divis) {
+		var currentX = Math.floor(coords.xPos/tSize)*tSize, //turns the continuous grid into discrete blocks
+			currentY = Math.floor(coords.yPos/tSize)*tSize,
+			tempX = currentX, 
+			tempY = currentY, 
+			update = 1, //as in change the dungeon path direction
+			tileCounter = 1, 
+			direction = Math.round(Math.random()*3), //used to determine the direction of the dungeon path
+			Nrand, 
+			rmCent; 
+		
+		points = [];
+		if(dungeonTiles.whereInStack([currentX, currentY])==-1) {
+			points.push([currentX, currentY]);
+			dungeonTiles.inStack.push([currentX, currentY]);
+		}
+
+		while (tileCounter < sizeOpts.tileLimit) {
+			//pick a direction to build in
+			switch(direction) {
+				case 0: { tempX += sizeOpts.tileSize; break; }
+				case 1: { tempX -= sizeOpts.tileSize; break; }
+				case 2: { tempY += sizeOpts.tileSize; break; }
+				case 3: { tempY -= sizeOpts.tileSize; break; }
+			}
+			posit = [tempX, tempY];
+			if(dungeonTiles.whereInStack(posit)==-1) {
```

Comment:
changing whereInStack to return some result, else false, would change this code to.
because any result other than the number 0 is truthy, and false is.....false
`if(dungeonTiles.whereInStack(posit)) {`